src/utils.ts

```diff
@@ -0,0 +1,147 @@
+// eslint-disable-next-line max-classes-per-file
+import { IAFile, IAMD, TimelineMoment } from './interfaces';
+
+export class Moment implements TimelineMoment {
+  altText: string;
+
+  desktopImg: string;
+
+  mobileImg: string;
+
+  link: string | undefined;
+
+  constructor() {
+    this.altText = '';
+    this.desktopImg = '';
+    this.mobileImg = '';
+    this.link = undefined;
+  }
+}
+
+export const shuffle = (list: any[]) => {
```

Comment:
This, my friend, is a great opportunity for [Generics](https://www.typescriptlang.org/docs/handbook/2/generics.html)! Since you're making a function that can shuffle an array that can be any type, we want to let the TS compiler know that it's going to return a list of that same type. Here's what a generic version of this function looks like:

```suggestion
export function shuffle<T>(list: T[]): T[] {
  const oldArray = [...list];
  let newArray: T[] = [];
```

`<T>` is just a placeholder for any type that might be passed in. We know the argument is going to be an array of type `T` and that we're going to return an array of type `T`. We also know the `newArray` is going to be of type `T` so now Typescript knows that all of these arrays are going to be of type `T`, but it doesn't care what `T` actually is.

Reply:
ah ok, so a named function would be better than anonymous =>?

Reply:
You might be able to use the `=>` syntax, I'm just not sure what the generic syntax for `=>` is.

Reply:
Ah I found it... looks like it'd be this:
```
export const shuffle = <T>(list: T[]): T[] => { }
```